utils/automatic-release/process.js

```diff
@@ -0,0 +1,121 @@
+    result += data.toString();
+  });
+
+  return new Promise(resolve => {
+    cmd.on('close', code => {
+      return resolve(
+        result
+          .split('\n')
+          .filter(line => line.match(/^-\sot/))
+          .map(line => line.replace(/^-\s/, ''))
+      );
+    });
+  });
+};
+
+// locally install latest version of packages
+const installLatestFromNPM = packages => {
+  fs.ensureDirSync(latestTemp);
+  fs.writeJsonSync(`${latestTemp}/package.json`, { name: 'release' });
+
+  const installed = packages.map(pkg => {
+    const cmd = spawn('npm', ['install', pkg], {
+      stdio: 'inherit',
+      cwd: latestTemp
+    });
+    return new Promise(resolve => {
+      cmd.on('close', code => (code === 0 ? resolve(pkg) : resolve(null)));
+    });
+  });
+
+  return installed;
+};
+
+const run = () => {
+  return new Promise((resolve, reject) => {
+    checkUPdated()
+      .then(updated => {
+        Promise.all(installLatestFromNPM(updated)).then(() => {
+          updated.forEach(pkg => {
+            const updatedPath = pkg.match(/^ottheme/)
+              ? `OTTheme/${pkg}/token.yml`
+              : `OTKit/${pkg}/token.yml`;
+            const updatedContent = yaml.load(path.join(root, updatedPath));
+
+            // If no version has been published yet, publish as major
+            let version = 'major';
+            let diff;
+            const latestPath = path.join(
+              latestTemp,
+              'node_modules',
+              pkg,
+              'token.yml'
+            );
+
+            if (fs.pathExistsSync(latestPath)) {
+              const latestContent = yaml.load(latestPath);
+              diff = differ(latestContent.props, updatedContent.props);
+              version = diffToSemver(latestContent.props, updatedContent.props);
+            }
+
+            if (version !== undefined) {
+              // spawn lerna publish --force-publish=package --cd-version=version --yes
+              // https://github.com/lerna/lerna#--force-publish-packages
+              // https://github.com/lerna/lerna#--yes
+              // https://github.com/lerna/lerna#--cd-version
+              console.log(`publishing ${pkg} as ${version}`);
+              const cmd = spawn(
+                'npm',
+                [
+                  'run',
+                  'publish',
+                  '--',
+                  `--force-publish=${pkg}`,
+                  `--cd-version=${version}`,
+                  '--yes'
+                ],
+                {
+                  cwd: root
+                }
+              );
+              cmd.on('error', err => {
+                console.log(`Error publishing ${pkg} as ${version}`, err);
+                throw err;
+              });
+            } else {
+              console.log(`Skipping publishing ${pkg}`);
+            }
+            console.log(diff);
+            console.log('--------------------------------');
+          });
+          fs.removeSync(latestTemp);
```

Comment:
should this be in a finally instead? So that we are sure it's performed in any case

Reply:
Added in case of error as well, until `.finally()` is avail on node (currently on stage4) - https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/finally